File: Binary_search.py
def bineary_search(array, value):
    left = 0
    right = len(array) - 1
    while left <= right:
        mid = (left + right) //2
        if array[mid] == value:
            return mid
        elif array[mid] < value:
            left = mid + 1
        else:
            right = mid - 1
    return -1
def verify(index):
    if index == -1:
        print("Element not found in the array.")
    else:
        print(f"Element found at index: {index}")

arr = [1, 2, 3, 4, 5]
tar = 3
result = bineary_search(arr, tar)
verify(result)

File: Binary_search.js
let list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];

function binarySearch(array, key){
    let left =0;
    let right = array.length - 1;
    while(left <= right){
        const mid = Math.floor((left + right) /2);
        console.log(`Checking middle index: ${mid}, value: ${array[mid]}`); // Log the current middle index and its value
        
        if(array[mid] === key){
            return console.log(`Element found at index: ${mid}`); // Return the index of the target element
        }
        if(array[mid] < key){
            left = mid + 1; // Search in the right half
        } else {
            right = mid - 1; // Search in the left half
        }
        console.log(`Searching in range: ${left} to ${right}`); // Log the current search range
    }
    return console.log(`Element not found`); // Return -1 if the target element is not found
}
binarySearch(list, 8); // Example usage, searching for the number 5

